src/App.tsx

```diff
@@ -14,11 +15,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isListVisible: boolean;
+  isReversed: boolean;
+  sortBy: 'alphabetically' | 'byLength' | null;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isListVisible: false,
+    isReversed: false,
+    sortBy: null,
+  };
+
+  showList = () => this.setState({ isListVisible: true });
+
+  reverseList = () => this.setState({ isReversed: true });
+
+  sortList = (sortOption: 'alphabetically' | 'byLength') => (
```

Comment:
again, enum would perfectly cover type definitions
```suggestion
  sortList = (sortOption: SortBy) => (
```